packages/fluentui/react-northstar/src/components/Tree/TreeTitle.tsx

```diff
@@ -116,6 +154,33 @@ const TreeTitle: React.FC<WithAsProp<TreeTitleProps>> & FluentComponentStaticPro
       })}
     >
       {childrenExist(children) ? children : content}
+      {// if tree item has subtree but it is not selectable explicitly, do not redner any selection indicator
+
+      selectable &&
+        hasSubtree &&
+        selectableParent &&
+        expanded &&
+        Box.create(selectionIndicator, {
+          defaultProps: () => ({
+            as: 'span',
+            ...getA11Props('indicator', {
+              className: TreeTitle.slotClassNames.indicator,
+              styles: resolvedStyles.customSelectionIndicator,
+            }),
+          }),
+        })}
+
+      {selectable &&
+        !hasSubtree &&
+        Box.create(selectionIndicator, {
```

Comment:
this is basically the same Component as above. Can't we just refactor the condition and use this element only once to avoid duplication?

Reply:
please review again, changes were applied by Charles :)